scripts/lb-wrapper

```diff
@@ -1,48 +1,50 @@
 #!/bin/bash
 
 LOG_FILE="/var/log/xklb.log"
+LOG_LEVEL="-vv"
 XKLB_EXECUTABLE="${XKLB_EXECUTABLE:-lb}"
 TMP_DOWNLOADS_DIR="/library/downloads/calibre-web"
 SURVEY_DB_FILE="${TMP_DOWNLOADS_DIR}/survey.db"
 URL="$1"
 FORMAT_OPTIONS="--format best --format-sort 'tbr~1000'"
-XKLB_FULL_CMD="${XKLB_EXECUTABLE} tubeadd ${SURVEY_DB_FILE} ${URL} --verbose && ${XKLB_EXECUTABLE} dl ${SURVEY_DB_FILE} --prefix ${TMP_DOWNLOADS_DIR} --write-thumbnail ${FORMAT_OPTIONS} --video ${URL} --verbose"
+XKLB_FULL_CMD="${XKLB_EXECUTABLE} tubeadd ${SURVEY_DB_FILE} ${URL} ${LOG_LEVEL} && ${XKLB_EXECUTABLE} dl ${SURVEY_DB_FILE} --prefix ${TMP_DOWNLOADS_DIR} --write-thumbnail ${FORMAT_OPTIONS} --video ${URL} ${LOG_LEVEL}"
 
 mkdir -p ${TMP_DOWNLOADS_DIR}
 
-# Function to log messages (and errors especially!)
+# Function to log messages with different levels
 log() {
-    echo "$(date +'%Y-%m-%d %H:%M:%S') - $1" | tee -a ${LOG_FILE}
+    local level=$1
+    local message=$2
+    echo "$(date +'%Y-%m-%d %H:%M:%S') - [$level] $message" >> ${LOG_FILE}
```

Comment:
Should the `tee` command be preserved, similar to the original Line 15?

(Or is there a very strong reason to remove it?)

Reply:
Yes, it's necessary to capture stdout of the xklb subprocess. But this will affect the log function, making output unecessary date and log level to empty lines...

I'll make an if---else to optimize it.

Reply:
If `tee` causes no material harm, why not preserve it?

Reply:
> If `tee` causes no material harm, why not preserve it?

I do preserve tee

> If `tee` causes no material harm, why not preserve it?

It does causes material harm writing empty lines to the log files while printing stdout.

Reply:
When did `tee` start causing harm?

(If this is new behavior, please share the link to the commit-or-code clarifying what is causing this new behavior, Thanks!)

Reply:
RELATED: `scripts/lb-wrapper` should definitely remain runnable from the CLI so everyone helping out here can build full confidence, in all moving parts separately (i.e. unit tests, prior to integration testing, is how QA works after all!)

So that probably means that... lb-wrapper needs to have its own STDOUT at minimum (and possibly also its own STDERR?)

Reply:
lb-wrapper already has its own STDOUT. Where do you think all "echo" commands in it print to?

Reply:
> When did `tee` start causing harm?
> 
> (If this is new behavior, please share the link to the commit-or-code clarifying what is causing this new behavior, Thanks!)

If you prefer `tee`, please provide a way to prevent logging empty lines into xklb.log. All those empty lines are considered as messages and are prepended with the date and log level by the log() function. I know it because I did tests with it.

Reply:
> When did `tee` start causing harm?

Can you please answer the actual above question?

(And if so: what design change or code change introduced this change in behavior?)

Reply:
When it prints yt-dlp process to xklb.log

Reply:
> When did `tee` start causing harm?

My # 1 is for a date please.

(Or are you saying that `tee` always caused harm?)

Reply:
This current version/design is the 3rd iteration I test so far:
1. Polling from a temporary file
2. Polling from stdout directly and write both to log and lb-wrapper stdout using tee
3. Polling from stdout directly and write both to log file and lb-wrapper stdout directly while filtering yt-dlp stdout

Reply:
> > When did `tee` start causing harm?
> 
> My # 1 question: on what date did `tee` start causing harm?
> 
> (Or are you saying that `tee` always caused harm?)

The term 'harm' is exaggerated. I just looked at my local history, can't find an exact snapshot of the problematic algorithm, but I'll reproduce it here as a commit for diff[ing] purpose.

Reply:
Here  you are, the date you are asking for is Jan 8, 2024, but like I said the term "harm" is exaggerated. Please refer to my previous messages explaining my expectations for "tee". Please also refer to https://github.com/deldesir/calibre-web/commit/6ec77ef04ade4e06578cf4512315d27bda723b59#diff-cf311048fe88bbcaabee62b916f61e84be86bbe4bcca2d54655b1b9efc66098cL15 to see how tee was used.

Reply:
I'm having trouble understanding you:

- Are you saying `tee` suddenly started causing problems on Jan 8?
   - If so, what specific underlying design change caused this problem?

- Or are you saying `tee` was in fact causing this problem (polluting logs) for many weeks?

Reply:
I used tee only yesterday in my offline tests because I wanted to write both to lb-wrapper stdout and xklb.log. I only used it yesterday. So to answer your questions:
 
>  * Are you saying `tee` suddenly started causing problems on Jan 8?

Not applicable because I was not using it Jan 7, Jan 6, Jan 5....

>  * If so, what specific underlying design change caused this problem?

No relevant changes worth mentioning, because it was not advertised as a functional code in some commits that stop to work suddenly by some unknown factors

> * Or are you saying `tee` was in fact polluting logs for many weeks?

Tee was polluting logs in my offline and private tests, only yesterday Jan 8.

Reply:
That helps a lot (convey intuition).

Towards understanding what your intent/purpose is.